boards/makerdiary/nrf52840_m2/nrf52840_m2.dts

```diff
@@ -0,0 +1,317 @@
+		           <6 0 &gpio0 15 0>, /* D0 */
+		           <7 0 &gpio0 16 0>, /* D1 */
+		           <8 0 &gpio0 19 0>, /* D2 */
+		           <9 0 &gpio0 20 0>, /* D3 */
+		           <10 0 &gpio0 21 0>, /* D4 */
+		           <11 0 &gpio0 22 0>, /* D5 */
+		           <12 0 &gpio0 23 0>, /* D6 */
+		           <13 0 &gpio0 24 0>, /* D7 */
+		           <14 0 &gpio1 0 0>, /* D8 */
+		           <15 0 &gpio1 1 0>, /* D9 */
+		           <16 0 &gpio1 2 0>, /* D10 */
+		           <17 0 &gpio1 3 0>, /* D11 */
+		           <18 0 &gpio1 4 0>, /* D12 */
+		           <19 0 &gpio1 7 0>, /* D13 */
+		           <20 0 &gpio1 5 0>, /* D14 */
+		           <21 0 &gpio1 6 0>; /* D15 */
+	};
+
+	arduino_adc: analog-connector {
+		compatible = "arduino,uno-adc";
+		#io-channel-cells = <1>;
+		io-channel-map = <0 &adc 0>, /* A0 = P0.02 = AIN0 */
+		                 <1 &adc 1>, /* A1 = P0.03 = AIN1 */
+		                 <2 &adc 4>, /* A2 = P0.28 = AIN4 */
+		                 <5 &adc 2>; /* A5 = P0.04 = AIN2 */
+	};
+
+	/* These aliases are provided for compatibility with samples */
+	aliases {
+		sw0 = &button0;
+		led0 = &led0_green;
+		led1 = &led1_red;
+		led2 = &led1_green;
+		led3 = &led1_blue;
+		led0-green = &led0_green;
+		led1-red   = &led1_red;
+		led1-green = &led1_green;
+		led1-blue  = &led1_blue;
+		pwm-led0 = &red_pwm_led;
+		pwm-led1 = &green_pwm_led;
+		pwm-led2 = &blue_pwm_led;
+		pwm-led3 = &backlight_pwm_led;
+		red-pwm-led = &red_pwm_led;
+		green-pwm-led = &green_pwm_led;
+		blue-pwm-led = &blue_pwm_led;
+		backlight-pwm-led = &backlight_pwm_led;
+		bootloader-led0 = &led0_green;
+		mcuboot-button0 = &button0;
+		mcuboot-led0 = &led0_green;
+		watchdog0 = &wdt0;
+		spi-flash0 = &mx25r64;
+	};
+};
+
+&adc {
+	status = "okay";
+};
+
+&gpiote {
+	status = "okay";
+};
+
+&gpio0 {
+	status = "okay";
+};
+
+&gpio1 {
+	status = "okay";
+};
+
+&uart0 {
+	compatible = "nordic,nrf-uart";
+	current-speed = <115200>;
+	status = "okay";
+	pinctrl-0 = <&uart0_default>;
+	pinctrl-1 = <&uart0_sleep>;
+	pinctrl-names = "default", "sleep";
+};
+
+arduino_i2c: &i2c0 {
+	compatible = "nordic,nrf-twi";
+	status = "okay";
+	pinctrl-0 = <&i2c0_default>;
+	pinctrl-1 = <&i2c0_sleep>;
+	pinctrl-names = "default", "sleep";
+};
+
+&i2c1 {
+	compatible = "nordic,nrf-twi";
+	status = "okay";
+	pinctrl-0 = <&i2c1_default>;
+	pinctrl-1 = <&i2c1_sleep>;
+	pinctrl-names = "default", "sleep";
+};
+
+&pwm0 {
+	status = "okay";
+	pinctrl-0 = <&pwm0_default>;
+	pinctrl-1 = <&pwm0_sleep>;
+	pinctrl-names = "default", "sleep";
+};
+
+arduino_spi: &spi1 {
+	compatible = "nordic,nrf-spi";
+	status = "okay";
+	pinctrl-0 = <&spi1_default>;
+	pinctrl-1 = <&spi1_sleep>;
+	pinctrl-names = "default", "sleep";
+	cs-gpios = <&arduino_header 16 0>; /* D10 */
+};
+
+
+&spi0 {
+	compatible = "nordic,nrf-spi";
+	status = "okay";
+	pinctrl-0 = <&spi0_default>;
+	pinctrl-1 = <&spi0_sleep>;
+	pinctrl-names = "default", "sleep";
+	cs-gpios = <&gpio0 6 0>,<&gpio0 7 0>;
+
+	st7789v@0 {
```

Comment:
might want to move these to spi3 and use `	compatible = "nordic,nrf-spim";` as spim3 is the fast interface that supports 32MHz instead of 8MHz

Reply:
Yes, I mentioned it in the PR description, that it could be changed. Manufacturer repo had it as is. I will change it.